File: python/nth-prime/nth_prime.py
def prime(positive_number):
    if positive_number <= 0:
        raise ValueError('Invalid input!')

    prime = 1
    for _ in range(positive_number):
        prime = next_prime(prime)
    return prime


def next_prime(x):
    while True:
        x += 1
        if is_prime(x):
            return x


def is_prime(x):
    i = 2
    while i * i <= x:
        if x % i == 0:
            return False
        i += 1
    return True

File: cpp/nth-prime/nth_prime.cpp
#include "nth_prime.h"
#include <stdexcept>

namespace nth_prime
{
bool is_prime(int x)
{
    for (int i{2}; i * i <= x; ++i)
    {
        if (x % i == 0)
        {
            return false;
        }
    }

    return true;
}

int next_prime(int x)
{
    while (true)
    {
        ++x;

        if (is_prime(x))
        {
            return x;
        }
    }
}

int nth(int n)
{
    if (n <= 0)
    {
        throw std::domain_error{"Invalid input!"};
    }

    int prime{1};
    for (int i{0}; i < n; ++i)
    {
        prime = next_prime(prime);
    }

    return prime;
}
} // namespace nth_prime
